geonode/qgis_server/views.py

```diff
@@ -740,3 +740,35 @@ def set_thumbnail(request, layername):
     }
     return HttpResponse(
         json.dumps(retval), content_type="application/json")
+
+
+def download_qlr(request, layername):
+    """Download QLR file for a layer
+
+    :param layername: The layer name in Geonode.
+    :type layername: basestring
+    :return: QLR file
+    """
+
+    layer = get_object_or_404(Layer, name=layername)
+    url = settings.OGC_SERVER['default']['PUBLIC_LOCATION'] + 'ogc/' + layername
```

Comment:
I think we shouldn't hardcode the URL here 'ogc/'. It's better to use the reverse function to get the full URL from 'urls.py'